server/broadcast.go

```diff
@@ -74,24 +80,57 @@ type BroadcastSessionsManager struct {
 func (bsm *BroadcastSessionsManager) selectSession() *BroadcastSession {
 	bsm.sessLock.Lock()
 	defer bsm.sessLock.Unlock()
+	numSess := bsm.sel.Size()
 
 	checkSessions := func(m *BroadcastSessionsManager) bool {
-		numSess := m.sel.Size()
 		if numSess < int(math.Ceil(float64(m.numOrchs)/2.0)) {
 			go m.refreshSessions()
 		}
-		return numSess > 0
+		return (numSess > 0 || bsm.lastSess != nil)
 	}
 	for checkSessions(bsm) {
-		sess := bsm.sel.Select()
-		// Handle the case where there is an error during selection and Select() returns nil when session list length > 0
+		var sess *BroadcastSession
+
+		if bsm.lastSess != nil && len(bsm.lastSess.SegsInFlight) > 0 &&
+			time.Now().Sub(bsm.lastSess.SegsInFlight[0].startTime) < bsm.lastSess.SegsInFlight[0].segDur {
+			// Re-use last session if oldest segment is in-flight for < segDur
+			glog.Warningf("REUSING OLD SESSION")
+			sess = bsm.lastSess
+		} else if numSess > 0 {
```

Comment:
Ah I encountered an error when `sel.Select()` was called on an empty list with `LIFOSelector` but I guess a better solution would be to add a check in it's `Select()` method and return nil - rather than adding a check during calling. Will do in the update and revert back the `numSess` changes.